server/plugin.go

```diff
@@ -247,6 +247,8 @@ func (p *Plugin) OnActivate() error {
 	ptr := p.client.Configuration.GetConfig().ServiceSettings.SiteURL
 	if ptr != nil {
 		mattermostSiteURL = *ptr
+	} else {
+		return errors.New("failed to read SiteURL configuration")
```

Comment:
This ends up being displayed on the plugin management page, so we should try to make this as actionable as possible. How about this:

```suggestion
		return errors.New("please configure the Mattermost server's SiteURL, then restart the plugin.")
```

Reply:
Thank you for your suggestion. This is in fact much better.